File: app/src/main/java/com/rakuishi/ok/model/Feed.java
package com.rakuishi.ok.model;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Path;
import org.simpleframework.xml.Root;

import java.util.List;

@Root(strict = false)
public class Feed {

    @Path("channel")
    @Element(name = "title")
    public String title;

    @Path("channel")
    @Element(name = "link")
    public String link;

    @Path("channel")
    @Element(name = "updated")
    public String updated;

    @Path("channel")
    @ElementList(inline = true)
    public List<FeedItem> list;
}

File: app/src/main/java/com/rakuishi/ok/data/Feed.kt
package com.rakuishi.ok.data

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "item", strict = false)
class Feed {
    @get:Element(name = "title")
    @set:Element(name = "title")
    var title: String = ""

    @get:Element(name = "link")
    @set:Element(name = "link")
    var link: String = ""

    @get:Element(name = "description")
    @set:Element(name = "description")
    var description: String = ""

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}